### PR TITLE
Swift 4 conversion

### PR DESCRIPTION
**Pull request by tzm41**
- Pods dependencies are pointing temporarily to Swift 4 beta releases of RxSwift. Podspec dependencies are not updated yet.
- `identifier_name` rule of SwiftLint disabled, unless we eventually decide against `ip_` prefixes.
- I think the necessity of `Integer+Extensions`, `UnsignedInteger+Extensions`, and `Mathable` could be reconsidered.

This PR is waiting on a stable 4.0 release of RxSwift. According to RxSwift discussions that would be mid-October.

### CODE REVIEW

**Comment by tzm41** [comment]
Waiting on https://github.com/SlatherOrg/slather/pull/321 to allow code coverage report be generated from Xcode 9 build.

**Comment by paulrolfe** [comment]
Looks like the slather fix got merged. I think this is ready 👍

**Comment by tzm41** [comment]
Sure, if we are ready for Swift 4

**Comment by tzm41** [comment]
Need to get more people to review and approve

**Comment by bobgilmore** [comment]
Do we need to add info the README to show how to point to the last good 3.x-compatible branch? And create said branch?

**Comment by paulrolfe** [comment]
@tzm41 Definitely need some more reviews and buy-in, agreed.

Good idea @bobgilmore, but doesn't need a new branch, we should just make it clear that `0.8.4` is the last swift 3 version.

**Comment by bobgilmore** [comment]
@paulrolfe Derp, of course. 

Not STRICTLY related to the PR, but worth thinking about - what tag are we planning on using for the Swift 4-compatible release? Not 0.8.5, right? Going to bump to 0.9 or something? Asking because I have a project that I'd like to lock to 3.x-compatible versions, and I'd like (I think) to use `~>` in the Podfile, not `< 0.8.5`

**Comment by IntrepidBuildServer** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The iPhone simluator specified in your Jenkinsfile is invalid for this version of Xcode</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/IntrepidPursuits/swift-wisdom/blob/b9314b2244c613812939307e4f9354446c1d09de/DispatchQueueConfiguration.swift#L46">DispatchQueueConfiguration.swift#L46</a> - <strong><a href="https://github.com/IntrepidPursuits/swift-wisdom/blob/b9314b2244c613812939307e4f9354446c1d09de/Pods/RxSwift/RxSwift/Schedulers/Internal/DispatchQueueConfiguration.swift#L46">Pods/RxSwift/RxSwift/Schedulers/Internal/DispatchQueueConfiguration.swift#L46</a></strong>: ‘scheduleOneshot(deadline:leeway:)’ is deprecated: renamed to ‘schedule(deadline:repeating:leeway:)’  <br /><code>
        timer.scheduleOneshot(deadline: deadline, leeway: leeway)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/IntrepidPursuits/swift-wisdom/blob/b9314b2244c613812939307e4f9354446c1d09de/DispatchQueueConfiguration.swift#L80">DispatchQueueConfiguration.swift#L80</a> - <strong><a href="https://github.com/IntrepidPursuits/swift-wisdom/blob/b9314b2244c613812939307e4f9354446c1d09de/Pods/RxSwift/RxSwift/Schedulers/Internal/DispatchQueueConfiguration.swift#L80">Pods/RxSwift/RxSwift/Schedulers/Internal/DispatchQueueConfiguration.swift#L80</a></strong>: ‘scheduleRepeating(deadline:interval:leeway:)’ is deprecated: renamed to ‘schedule(deadline:repeating:leeway:)’  <br /><code>
        timer.scheduleRepeating(deadline: initial, interval: dispatchInterval(period), leeway: leeway)
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 167 tests, with 0 failures (0 unexpected) in 0.615 (0.751) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Jenkinsfile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 def xcodeProject = new io.intrepid.XcodeProject()
 xcodeProject.name = 'SwiftWisdom'
 xcodeProject.addBuild([ configuration: "Debug" ])
+xcodeProject.xcodeVersion = '9'
```
Comment (aspitz-intrepid) [inline Jenkinsfile]: might want to add `xcodeProject.simulator = 'iPhone 7 (11.0)'` just to be on the safe side

File: Podfile [hunk 2]
```diff
@@ -6,8 +6,8 @@ use_frameworks!
 
 def commonpods
   pod 'IP-UIKit-Wisdom'
-    pod 'RxSwift', '~> 3.0'
-    pod 'RxCocoa', '~> 3.0'
+    pod 'RxSwift', :git => 'https://github.com/ReactiveX/RxSwift.git', :branch => 'swift4.0'
+    pod 'RxCocoa', :git => 'https://github.com/ReactiveX/RxSwift.git', :branch => 'swift4.0'
```
Comment (Tom-Le) [inline Podfile]: Is this the right branch to point to? [RxSwift's README](https://github.com/ReactiveX/RxSwift) is pointing to a different branch for Swift 4. Also [CocoaPods' RxSwift page](https://cocoapods.org/?q=rxswift) says the latest version is 4.0.0-beta.0 which is Swift 4 compatible (I just tested); should we point directly to that instead?

Comment (tzm41) [reply Podfile]: Your point looks valid. I think they changed their branch between then and now.

Comment (Tom-Le) [reply Podfile]: :classic:

Comment (tzm41) [reply Podfile]: Damn, they haven't pushed `RxTest` 4 version.

Comment (tzm41) [reply Podfile]: Wait no it works

**Comment by IntrepidBuildServer** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The iPhone simluator specified in your Jenkinsfile is invalid for this version of Xcode</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 167 tests, with 0 failures (0 unexpected) in 0.563 (0.657) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 167 tests, with 0 failures (0 unexpected) in 0.563 (0.657) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by IntrepidBuildServer** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 167 tests, with 0 failures (0 unexpected) in 0.671 (0.782) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Podfile [hunk 3]
```diff
@@ -1,13 +1,14 @@
 source 'https://github.com/CocoaPods/Specs'
 
-platform :ios, '8.0'
+platform :ios, '10.0'
```
Comment (paulrolfe) [inline Podfile]: What's the reason for bumping this? I think we should support as many iOS versions as possible.

Comment (tzm41) [reply Podfile]: You’re right there’s probably no reason to bump it. But the podspec target isn’t changed either so I assume user of library won’t be affected by this?

Comment (paulrolfe) [reply Podfile]: Yeah, you're probably right. You can keep it then, as long as Intrepid consumers are not limited by this iOS version.

**Review by bobgilmore — APPROVED** [review]
I think it's a LOVELY PR and you should merge it! :shipit:

**Comment by paulrolfe** [comment]
@bobgilmore I value and respect your feedback and approval.

However, `pod spec lint` is failing, so I think we do need some changes... investigating now 😁

**Comment by tzm41** [comment]
Our team is definitely very respectful. Thanks guys!

**Comment by IntrepidBuildServer** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 167 tests, with 0 failures (0 unexpected) in 0.539 (0.688) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by tzm41** [comment]
This PR is waiting on a stable 4.0 release of RxSwift. According to RxSwift discussions that would be mid-October.

File: Podfile (hunk 3 above)
Comment (Benuuu) [reply Podfile]: This affects our test application, we should keep it at our minimum supported platform.

File: SwiftWisdom.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -834,16 +834,18 @@
...
 			attributes = {
 				LastSwiftUpdateCheck = 0710;
-				LastUpgradeCheck = 0810;
+				LastUpgradeCheck = 0900;
 				ORGANIZATIONNAME = Intrepid;
 				TargetAttributes = {
 					80232B9E1BF2EE2E00818B6E = {
 						CreatedOnToolsVersion = 7.1;
-						LastSwiftMigration = 0800;
+						DevelopmentTeam = THP5EV5EJ9;
```
Comment (Benuuu) [inline SwiftWisdom.xcodeproj/project.pbxproj]: Did the development team need to sneak in here?

**Review by Benuuu — APPROVED** [review]

**Comment by IntrepidBuildServer** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 167 tests, with 0 failures (0 unexpected) in 0.406 (0.484) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Review by alexpersian — APPROVED** [review]

**Review by aspitz-intrepid — APPROVED** [review]
Go go go

**Comment by tzm41** [comment]
Hmm this is RC.. Are we good to go?

**Review by paulrolfe — APPROVED** [review]

**Comment by ghost** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 167 tests, with 0 failures (0 unexpected) in 0.556 (0.655) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**